source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -128,9 +128,14 @@ void RedisCluster::DnsDiscoveryResolveTarget::startResolve() {
 
   active_query_ = parent_.dns_resolver_->resolve(
       dns_address_, parent_.dns_lookup_family_,
-      [this](std::list<Network::Address::InstanceConstSharedPtr>&& address_list) -> void {
+      [this](const std::list<Network::DnsResponse>& response) -> void {
         active_query_ = nullptr;
         ENVOY_LOG(trace, "async DNS resolution complete for {}", dns_address_);
+        std::list<Envoy::Network::Address::InstanceConstSharedPtr> address_list;
```

Comment:
Can you just alter the session code to accept the new DNS response list via move semantics? The TTL data can come with it in case we want to use it in the future.

Reply:
Sure. I will change it.